docs/reference/cluster/allocation-explain.asciidoc

```diff
@@ -180,6 +180,8 @@ primary shard that was previously allocated.
 ----
 // NOTCONSOLE
 
+The shard in the example above errors `no_valid_shard_copy` due to `NODE_LEFT`. The first step to recover is to make sure all nodes are in the cluster. Then, if the error continues after a <<cluster-reroute,cluster reroute>>, the data will need to be <<snapshots-restore-snapshot,restored from snapshot>>.
```

Comment:
The `NODE_LEFT` bit is a distraction - it describes why the shard became unassigned in the first place, but that's not normally relevant to the process of working out how to recover. I'd suggest not mentioning it.

We definitely shouldn't suggest performing a cluster reroute either. If that's needed, it'll say so explicitly in the API response. It doesn't say so in the example, so it won't be needed. We might at some point actually reject such unnecessary calls, see https://github.com/elastic/elasticsearch/issues/81231 for more information.

More generally, I think the best place for this sort of advice is the [relevant troubleshooting page](https://www.elastic.co/guide/en/elasticsearch/reference/current/red-yellow-cluster-status.html). That covers a lot more cases than just this one example. Can we just link to that page instead?

Reply:
ACK on just linking the other page; sounds like we're discussing in thread above, cheers. 

I will highlight for the example, `NODE_LEFT` is applicable for the caveat "make sure all nodes are in the cluster" because that's the first resolution step and a common user gotcha IMO. 

However, agreed we shouldn't necessarily need a Cluster Reroute, Support's just used to recommending that to get the `retry_failed=true` option for the cluster to reconsider faster. (Or it does nothing but it seems to do something 🙂 .)

Reply:
perhaps this could be a tip with links, like the following:

```suggestion
TIP: If a shard is unassigned with an allocation status of `no_valid_shard_copy`, then you should <<fix-cluster-status-recover-nodes,make sure that all nodes are in the cluster>>. If the node containing a primary shard is lost, then you can <<fix-cluster-status-restore,recover the data for the shard>>.
```